File: One Edit Distance.java
class Solution {
    public boolean isOneEditDistance(String s, String t) {
        int longL = s.length();
        int shortL = t.length();
        
        if (shortL > longL){
            return isOneEditDistance(t ,s);
        }
        
        if (longL - shortL > 1){
            return false;
        }
        
        for (int i = 0; i < shortL; i ++) {
            if (s.charAt(i) != t.charAt(i)){
                
                if (shortL == longL){
                    return s.substring(i + 1).equals(t.substring (i + 1));
                } else {
                    return s.substring(i + 1).equals(t.substring(i) );
                }
                
            }
        }
        
        return longL == shortL + 1;
    }
}

File: One Edit Distance.py
class Solution:
    def isOneEditDistance(self, s: str, t: str) -> bool:
        if abs( len(s) - len(t) ) > 1:
            return False; 
        
        m , n = len(s), len(t)
        if m >= n:
            longer = s;
            shorter = t;
        else:
            longer = t;
            shorter = s;
            
        for i in range(len(shorter)):
            if longer[i] != shorter[i]:
                
                if m == n:
                    return longer[i + 1:]  == shorter[i + 1:];
                
                else:
                    
                    return longer[i + 1:] == shorter[i:];
                
                
        return len(longer) - len(shorter) == 1;
    
    
    
        